tox.ini

```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py35,py36,py37,py38,flake
+envlist = py27,py35,py36,py37,py38,py39,py310,py311,py312,flake
```

Comment:
Would it be worth dropping the [EOL'd Python versions](https://devguide.python.org/versions/)? For the record, I don't have anything against ensuring the project keeps running in old versions, as long as that doesn't compromise our ability to adopt new features or cleanup the codebase.

Reply:
See comment in main thread - I agree, but kinda planned to do that later. I'm going to suggest dropping Python 2 support entirely, and along with that I'd usually do some stuff like removing any now-unnecessary compat stuff for Python 2 (e.g. use of six), switching to format strings...

Reply:
gonna mark this as resolved and we'll address it later.